lib/OIDCLogoutHandler.php

```diff
@@ -70,4 +70,21 @@ public function handleResponseFromRequest(array $request)
         \SimpleSAML\Auth\Source::completeLogout($state);
         // @codeCoverageIgnoreStart
     }
+
+    public function handleRequest()
+    {
+        $this->handleRequestFromRequest($_REQUEST);
+    }
+
+    public function handleRequestFromRequest(array $request)
```

Comment:
Why is this necessary? If we need the additional level of indirection, maybe `handleRequestFromRequest()` should be private?

If the reason for the extra method is to have a default, I would do that in the `handleRequest()` method. We cannot use a variable as a default value for a function argument, but we can default it to `null` and capture that in the code:

```php
    public function handleRequest(array $request = null)
    {
        $request = $request ?: $_REQUEST;

        Logger::debug('authoauth2: logout request=' . var_export($request, true));
        $config = \SimpleSAML\Configuration::getInstance();
        $sourceId = $request['authSource'];
        $as = new \SimpleSAML\Auth\Simple($sourceId);
        $as->logout([
            'oidc:localLogout' => true,
            'ReturnTo' => $config->getBasePath().'logout.php',
        ]);
    }
```

It might be worth also to investigate the possibility of using `Symfony\Component\HttpFoundation\Request` objects instead of the `array` scalar type for the argument, although this depends heavily on the module and Patrick should have a better overview. The `symfony/http-foundation` package has been available as a dependency starting on SSP 1.17.0.

Reply:
The reason is to make it easier to write unit tests. Also, this is the way it was done for existing, similar code in the module.

Reply:
Uhm, I see... But wouldn't my suggestion make it possible to test as well? Then you have two options: build a request and pass it as an argument to `handleRequest()`, or take that same request and assign it to `$_REQUEST`.

Reply:
Indeed it would. I was just following existing code patterns in the module.